content/6_Advanced/SCC.mdx

```diff
@@ -24,13 +24,14 @@ strongly connected component. We can compute these components using either
 <Resources>
 	<Resource source="CPH" title="17 - Strong Connectivity" />
 	<Resource source="Wikipedia" title="Kosaraju's Algorithm" url="https://en.wikipedia.org/wiki/Kosaraju%27s_algorithm" />
+	<Resource source="TC" title="Kosaraju's Algorithm for SCCs"
+	url="https://www.topcoder.com/thrive/articles/kosarajus-algorithm-for-strongly-connected-components"/>
 </Resources>
 
-<IncompleteSection>
-
-in-house explanation?
+## Explanation
+Kosaraju's algorithm is an efficient method for finding the strongly connected components (SCCs) of a directed grpahs. The algorithm works in two main phase. In first phase, it performs the DFS on the original graphs $G$ and record the exit time (i.e. reversed topological sort of graph $G$).
 
-</IncompleteSection>
+In the second phase, the graph is transposed, meaning all the edges are reversed. A new DFS is performed, but this time it processes nodes in the order defined by the topological sort from the first phase. By reversing the edges and processing nodes in this specific order, each DFS run in this phase will identify all vertices of one SCC before moving to the next SCC. This is because in the transposed graph, the SCCs are still strongly connected, and the exit times help in isolating them during the DFS.
```

Comment:
``` This is because in the transposed graph, the SCCs are still strongly connected, and the exit times help in isolating them during the DFS.```

can you run a quick proof/example why?

Reply:
because in transposed graph the reachability will not change and the exit times we calculated from first dfs will help to process one entire SCC in second dfs before moving to another SCC.
For Proof : TopCoder Blog

Reply:
"TopCoder Blog" can you, like, actually give the url 💀

Reply:
https://www.topcoder.com/thrive/articles/kosarajus-algorithm-for-strongly-connected-components

Reply:
can you include a brief explanation yourself in addition to linking the blog?

Reply:
The strongly connected components (SCCs) are preserved in the transposed graph because reversing the edges of a strongly connected component does not change the fact that every vertex can reach every other vertex in the same component. This is based on the definition of an SCC: a maximal subgraph where every vertex is reachable from every other vertex in the subgraph. When we transpose the graph, all paths within each SCC are reversed. For example, if there was a path from A to B in the original graph, in the transposed graph there will be a path from B to A. So if every vertex could reach every other vertex in the original SCC, they can still reach each other in the transposed graph, just in reverse. The exit times recorded in the first phase are crucial in isolating SCCs in the second phase. In the first phase, we do a DFS on the original graph and record the exit times of the vertices, pushing them onto a stack as we finish processing them. This ensures that the vertices are in order of their completion times, with the vertex that finishes last (having the highest exit time) on top of the stack. 

In the second phase, we process the vertices in the order defined by this stack, which means we process them in decreasing order of their exit times. The vertex with the highest exit time is likely to be in a “sink” SCC in the transposed graph, meaning it can only be reached by other vertices but has no outgoing edges that leave its SCC. When we start a DFS from this vertex in the transposed graph we will explore all vertices in its SCC before we reach any vertex outside of it. This is because all vertices in the SCC are mutually reachable and the DFS will fully explore this strongly connected subgraph before exiting. By processing the vertices in this order we ensure that each DFS in the second phase is confined to a single SCC. Once all vertices in one SCC are processed we move to the next highest exit time vertex that hasn’t been visited yet and start the process for the next SCC.

Reply:
will it work for the "WHY" part?

Reply:
Not sure if this is just me, but some parts of this explanation feel wordy (?)
Overall, content looks fine though 👍 

Like
> The strongly connected components (SCCs) are preserved in the transposed graph because reversing the edges of a strongly connected component does not change the fact that every vertex can reach every other vertex in the same component. This is based on the definition of an SCC: a maximal subgraph where every vertex is reachable from every other vertex in the subgraph. When we transpose the graph, all paths within each SCC are reversed. For example, if there was a path from A to B in the original graph, in the transposed graph there will be a path from B to A. So if every vertex could reach every other vertex in the original SCC, they can still reach each other in the transposed graph, just in reverse.

could probably be reworded as:
> The transposed graph will have the same sccs as the original because reversing an edge doesn't change the reachability between two nodes. Since each scc is maximal and disjoint, reverse edges don't break any connections in the original graph. [example]

Reply:
Okay I will try to make it a bit concise

Reply:
Done.
Can you check it once @ryanchou-dev